modules/nw-metallb-configure-vrf-bgppeer.adoc

```diff
@@ -0,0 +1,179 @@
+
+.Procedure
+
+. Create `BGPPeer` and `BFDProfile` custom resources (CR).
+
+.. Create a file, such as `frrviavrf.yaml`, with content like the following example:
++
+[source,yaml]
+----
+---
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: frrviavrf
+  namespace: metallb-system
+spec:
+  myASN: 100
+  peerASN: 200
+  peerAddress: 192.168.130.1
+  vrf: ens4vrf <1>
+  bfdProfile: testprofile
+---
+apiVersion: metallb.io/v1beta1
+kind: BFDProfile <2>
+metadata:
+  name: testprofile
+  namespace: metallb-system
+----
+<1> Defines the VRF instance to associate with the BGP peer. MetalLB can advertise services and make routing decisions based on the routing information in the VRF.
+<2> Use a BFD profile for the BGP sessions for more rapid failure detection in routes.
+
+.. Apply the configuration for the BGP peer and BFD profile by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f frrviavrf.yaml
+----
+
+. Create an `IPAddressPool` CR.
+
+.. Create a file, such as `first-pool.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: first-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - 192.169.10.0/24
+----
+
+.. Apply the configuration for the IP address pool by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-pool.yaml
+----
+
+. Create a `BGPAdvertisement` CR.
+
+.. Create a file, such as `first-adv.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: first-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+    - first-pool
+  peers:
+    - frrviavrf <1>
+  nodeSelectors:
+    - matchLabels:
+        egress-service.k8s.ovn.org/test-server1: "" <2>
+----
+<1> In this example, MetalLB advertises a range of IP addresses from the `first-pool` IP address pool to the `frrviavrf` BGP peer. 
+<2> Optional: To use the same return path for the traffic originating from the pod, specify the node to advertise as the next hop for the service. 
+
+.. Apply the configuration for the BGP advertisement by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-adv.yaml
+----
+
+. Create a `Namespace`, `Deployment` and `Service` CR.
+
+.. Create a file, such as `deploy-service.yaml`, with content like the following example:
++
+[source,yaml,options="nowrap",role="white-space-pre"]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: test
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: server
+  namespace: test
+spec:
+  selector:
+    matchLabels:
+      app: server
+  template:
+    metadata:
+      labels:
+        app: server
+    spec:
+      containers:
+      - name: server
+        image: quay.io/centos7/httpd-24-centos7 
```

Comment:
Just double check what we suggest in the other examples

Reply:
Using `registry.redhat.io/rhel9/httpd-24` might be more appropriate but will check further later.